### PR TITLE
Project 11 - Petra Almgren

### PR DESCRIPTION
**Pull request by petrame**
Link to my deployed project:
https://project-happy-thoughts.netlify.app/

My reflections on how this week's project turned out:
Very good in the end even though it was a struggle to get all the pieces in place.

### CODE REVIEW

**Comment by petrame** [comment]
New pull request since I did some changes after a 1on1.

File: code/src/components/Footer.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  return (
+    <footer>
+      <p>
+        &copy;
+        <a href="https://github.com/petrame/project-happy-thoughts">{author}</a>
+        &nbsp;for Technigo Bootcamp 2020.
+      </p>
+    </footer>
+  );
+};
```
Comment (stjernberg) [inline code/src/components/Footer.js:15]: Nice with props in Footer!

File: code/src/components/MessageInput.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          💜
+        </span>
+        &nbsp;Send Happy Thought&nbsp;
+        <span className="button-heart" role="img" aria-label="heart">
+          💜
+        </span>
+      </button>
+    </form>
+  );
+};
```
Comment (stjernberg) [inline code/src/components/MessageInput.js:45]: Nice indentation and good that you split the code into different lines. That goes for all your component files.

File: code/src/components/MessageList.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          <div className="message-info">
+            <button
+              className={
+                message.hearts > 0
+                  ? "like-button with-likes"
+                  : "like-button no-likes"
+              }
+              onClick={() => handleSubmit(message._id)}
+            >
+              <span className="heart" role="img" aria-label="Click to like the post.">
```
Comment (stjernberg) [inline code/src/components/MessageList.js:27]: This was relly good aria label for the screen reader.

File: code/src/components/header.css [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  h1 {
+    margin: 30px 0 40px 0;
+    font-size: 64px;
+    transition: all 500ms;
+  }
+
+  h1:hover {
+    letter-spacing: 2px;
+  }
+}
```
Comment (stjernberg) [inline code/src/components/header.css:26]: Great that you splite the css files, looks much cleaner and are easy to read.!

File: code/src/components/loader.css [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+@keyframes spin {
+  0%,
+  15% {
+    transform: rotate(0);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+}
```
Comment (stjernberg) [inline code/src/components/loader.css:103]: Iäm mpresed with the loader styling you added. Good job!

File: code/src/components/MessageList.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import "./messageList.css";
+
+export const MessageList = ({ messages, onLiked }) => {
+  const handleSubmit = (id) => {
+    onLiked(id);
+  };
+
+  // Render messages using map
+  return (
+    // Add a section for each message returned by the backend
```
Comment (stjernberg) [inline code/src/components/MessageList.js:13]: I like that you comment what the function does. makes it easy to follow.

**Review by stjernberg — COMMENTED** [review]
You've done an awesome job with the Happy messages  👏. I really liked that you styled it your own way. I especially liked the striped noteses and the "pulsating" heart. Your code looks really good to me. it's structered and easy to follow with good naming on elements and comments on functions. It's well structured with the "waterfall" approach.  Well done 🤩

**Review by stjernberg — APPROVED** [review]

File: code/src/App.js [hunk 7]
```diff
@@ -1,9 +1,68 @@
...
+  }, []);
+
+  const fetchMessages = () => {
+    fetch(MESSAGE_URL)
+      .then((res) => res.json())
+      .then((data) => {
+        setTimeout(() => {
+          setLoader(false);
+          setMessages(data);
+        }, 3000);
```
Comment (EmOlsz) [inline code/src/App.js:27]: Not sure why you need 3000 ms to execute your code - it's not necessary at all

File: code/src/components/Loader.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+
+import "./loader.css";
+
+export const Loader = () => {
```
Comment (EmOlsz) [inline code/src/components/Loader.js:5]: Great loader, congrats :)

File: code/src/components/loader.css [hunk 9]
```diff
@@ -0,0 +1,103 @@
+.loader {
+  bottom: 0;
+  left: 0;
```
Comment (EmOlsz) [inline code/src/components/loader.css:3]: Those styles are not going to work without `position: absolute`

File: code/src/App.js (hunk 7 above)
Comment (petrame) [reply code/src/App.js:27]: It's just because I was so happy with the rainbow animation! :D

File: code/src/components/loader.css (hunk 9 above)
Comment (petrame) [reply code/src/components/loader.css:3]: True, remainings from trial and error. Removed now.

**Pull request closed without merging**